src/renderer/app-update-warning/__tests__/app-update-warning.test.ts

```diff
@@ -0,0 +1,169 @@
+      ["2022-06-05T05:31:00.000Z"],
+      ["2022-06-06T05:31:00.000Z"],
+      ["2022-06-07T05:31:00.000Z"],
+      ["2022-06-08T05:31:00.000Z"],
+      ["2022-06-09T05:31:00.000Z"],
+      ["2022-06-10T05:31:00.000Z"],
+      ["2022-06-11T05:31:00.000Z"],
+      ["2022-06-12T05:31:00.000Z"],
+      ["2022-06-13T05:31:00.000Z"],
+      ["2022-06-14T05:31:00.000Z"],
+      ["2022-06-15T05:31:00.000Z"],
+      ["2022-06-16T05:31:00.000Z"],
+      ["2022-06-17T05:31:00.000Z"],
+      ["2022-06-18T05:31:00.000Z"],
+      ["2022-06-19T04:31:00.000Z"],
+    ];
+
+    const mediumDates = [
+      ["2022-06-19T05:31:00.000Z"],
+      ["2022-06-20T05:31:00.000Z"],
+      ["2022-06-21T05:31:00.000Z"],
+      ["2022-06-22T05:31:00.000Z"],
+      ["2022-06-23T05:31:00.000Z"],
+      ["2022-06-24T04:31:00.000Z"],
+    ];
+
+    const highDates = [
+      ["2022-06-24T05:31:00.000Z"],
+      ["2022-06-25T05:31:00.000Z"],
+      ["2027-05-28T05:31:00.000Z"],
+    ];
+
+    describe.each(lightDates)("given AppUpdateWarning with update date set to %s", (date) => {
+      beforeAll(() => {
+        jest.useFakeTimers("modern");
+        jest.setSystemTime(new Date(date).getTime());
+      });
+
+      afterAll(() => {
+        jest.useRealTimers();
+      });
+
+      it(`returns light warning level if date is ${date}`, () => {
+        expect(appUpdateWarning.warningLevel).toBe("light");
+      });
+    });
+
+    describe.each(mediumDates)("given AppUpdateWarning with update date set to %s", (date) => {
+      beforeAll(() => {
+        jest.useFakeTimers("modern");
+        jest.setSystemTime(new Date(date).getTime());
+      });
+
+      afterAll(() => {
+        jest.useRealTimers();
+      });
+
+      it(`returns medium warning level if date is ${date}`, () => {
+        expect(appUpdateWarning.warningLevel).toBe("medium");
+      });
+    });
+
+    describe.each(highDates)("given AppUpdateWarning with update date set to %s", (date) => {
+      beforeAll(() => {
+        jest.useFakeTimers("modern");
+        jest.setSystemTime(new Date(date).getTime());
+      });
+
+      afterAll(() => {
+        jest.useRealTimers();
+      });
+
+      it(`returns medium warning level if date is ${date}`, () => {
+        expect(appUpdateWarning.warningLevel).toBe("high");
+      });
+    });
+  });
+
+  describe("given AppUpdateWarning with warning level changing over time", () => {
+    let appUpdateWarning: AppUpdateWarning;
+
+    beforeEach(() => {
+      jest.useFakeTimers("modern");
+      jest.setSystemTime(new Date("2022-05-30T05:30:00.000Z").getTime());
+      jest.spyOn(global, "setInterval");
+
+      appUpdateWarning = di.inject(appUpdateWarningInjectable);
+      appUpdateWarning.init();
+    });
+    
+    afterEach(() => {
+      jest.useRealTimers();
+    });
+
+    it("calls setInterval with correct arguments", () => {
+      const onceADay = 1000 * 60 * 60 * 24;
+
+      expect(setInterval).toHaveBeenCalledTimes(1);
```

Comment:
If there's already fake timers, why not test this realistically by advancing it, instead of spying it?